### PR TITLE
Allow for prepending to the cutbuffer in addition to replacing it.

### PR DESCRIPTION
**Pull request by toddpratt**

### CODE REVIEW

File: src/fe-text/gui-entry.h [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #ifndef __GUI_ENTRY_H
 #define __GUI_ENTRY_H
 
+#define CUTBUFFER_PREPEND 42
```
Comment (dequis) [inline src/fe-text/gui-entry.h]: I guess this line is redundant now

**Comment by dequis** [comment]
Welcome back! Just skimmed but I like what i see so far.

File: src/fe-text/gui-entry.c [hunk 2]
```diff
@@ -554,17 +554,39 @@ void gui_entry_erase(GUI_ENTRY_REC *entry, int size, int update_cutbuffer)
...
+			if (entry->cutbuffer_len) {
+				int cutbuffer_new_size = entry->cutbuffer_len + size;
+				unichar *tmpcutbuffer = entry->cutbuffer;
+				entry->cutbuffer = g_new(unichar, cutbuffer_new_size+1);
+
+				memcpy(entry->cutbuffer, entry->text + entry->pos - size,
+						size * sizeof(unichar));
+				memcpy(entry->cutbuffer + size, tmpcutbuffer,
+
+				entry->cutbuffer_len * sizeof(unichar));
```
Comment (dequis) [inline src/fe-text/gui-entry.c]: Indentation got really weird in this line.

File: src/fe-text/gui-entry.c [hunk 3]
```diff
@@ -624,7 +646,6 @@ void gui_entry_erase_word(GUI_ENTRY_REC *entry, int to_space)
 	}
 	if (to > 0) to++;
 
-        gui_entry_erase(entry, entry->pos-to, TRUE);
```
Comment (dequis) [inline src/fe-text/gui-entry.c]: What's up with this line being deleted? Mistake or somehow intentional?

Comment (toddpratt) [inline src/fe-text/gui-entry.c]: It was not intentional.  Fixed.

**Comment by ailin-nemui** [comment]
it should be extended with appending to the cutbuffer as well

**Comment by ailin-nemui** [comment]
@toddpratt I'll merge this now, I expect your follow ups

**Pull request merged**